src/5.1.no_vinnigrette.py

```
@@ -0,0 +1,58 @@
+"""
+This module provides date-related functions.
+Functions:
+- is_valid_date(date_str): Checks if the date is in a valid format (YYYY-MM-DD).
+- no_vinaigrette(date1, date2): Generates a random date between two given dates and checks if the date is a Wednesday.
+ If it's a Wednesday, it prints "I don't have vinegar".
+ Otherwise, it prints "Ain't gettin' no vinaigrette today :(".
+  Returns the random date and the corresponding day of the week.
+"""
+import datetime
+import random
+
+
+TIME_FORMAT = "%Y-%m-%d"
+
+def is_valid_date(date_str):
+    """Check if the date is in a valid format (YYYY-MM-DD)."""
+    try:
+        datetime.datetime.strptime(date_str, TIME_FORMAT)
+        return True
+    except ValueError:
+        return False
+
+def no_vinnigrete(date1, date2):
+    """Generate a random date between two given dates and check if the date is a Monday.
+    If it is, print a message saying "Ain't gettin' no vinaigrette today :(".
+    """
+    # Check if the date format is valid
+    if not is_valid_date(date1) or not is_valid_date(date2):
+        print("Invalid date format")
```

Return error instead of print and return none

-1